### PR TITLE
nom-sql: Parse fractional seconds precision on MySQL CURRENT_TIMESTAMP function

### PR DESCRIPTION
**Pull request by cameronbraid**
Fixes https://github.com/readysettech/readyset/issues/56

### CODE REVIEW

File: nom-sql/src/column.rs [hunk 1]
```diff
@@ -181,19 +187,23 @@ fn default(
...
+        let (i, _) = whitespace1(i)?;
+        let (i, _) = tag_no_case("update")(i)?;
+        let (i, _) = whitespace1(i)?;
+        let (i, _) = alt((
+            tag_no_case("current_timestamp"),
+            tag_no_case("now"),
+            tag_no_case("localtime"),
+            tag_no_case("localtimestamp"),
+        ))(i)?;
+        let (i, expr) = opt(delimited(tag("("), expression(dialect), tag(")")))(i)?;
```
Comment (glittershark) [inline nom-sql/src/column.rs:204]: I think this should probably accept whitespace before and around the parentheses

File: nom-sql/src/column.rs [hunk 2]
```diff
@@ -88,7 +88,7 @@ pub enum ColumnConstraint {
     Unique,
     /// NOTE(grfn): Yes, this really is its own special thing, not just an expression - see
     /// <https://dev.mysql.com/doc/refman/8.0/en/timestamp-initialization.html>
-    OnUpdateCurrentTimestamp,
+    OnUpdateCurrentTimestamp(Option<Expr>),
```
Comment (glittershark) [inline nom-sql/src/column.rs:91]: it doesn't seem like MySQL accepts an arbitrary expression here, only an integer:

```sql
MySQL [none]> CREATE TABLE t1 (
    ->   ts TIMESTAMP(6) DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(5 + 1)
    -> );
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '+ 1)
)' at line 2
```

**Review by glittershark — CHANGES_REQUESTED** [review]
Awesome, thanks for the PR! think we need to be a little more restrictive in the AST here to match what MySQL accepts

File: nom-sql/src/column.rs (hunk 2 above)
Comment (cameronbraid) [reply nom-sql/src/column.rs:91]: Agreed, however I was just copying the same pattern that is used for parsing `DEFAULT CURRENT_TIMESTAMP(6)` - which accepts `DEFAULT CURRENT_TIMESTAMP(5+1)`

Comment (cameronbraid) [reply nom-sql/src/column.rs]: I've restricted the parser to accepting `literal` and optional whitespace and updated test case accordingly

**Comment by cameronbraid** [comment]
Update fixes both review issues

File: nom-sql/src/column.rs (hunk 1 above)
Comment (cameronbraid) [reply nom-sql/src/column.rs]: done

**Comment by glittershark** [comment]
Sorry for the delay, lost track of this! Merged in https://github.com/readysettech/readyset/commit/669dd4e7e3c7faac1f8de8fa6f510b3366cd2df1

**Pull request closed without merging**